File: src/Sales.FSharp/BrewUp.Sales.Facade.FSharp/ISalesFacade.fs
namespace BrewUp.Sales.Facade.FSharp

open System.Threading.Tasks
open BrewUp.Shared.FSharp.ExternalContracts
open BrewUp.Shared.FSharp.ReadModel

type ISalesFacade =
    abstract CreateSalesOrderAsync: CreateSalesOrderJson -> Task<string>
    abstract GetSalesOrdersAsync: page: int * pageSize: int -> Task<PagedResult<SalesOrderJson>>

File: src/Sales/BrewUp.Sales.Facade/ISalesFacade.cs
using BrewUp.Shared.ExternalContracts;
using BrewUp.Shared.ReadModel;

namespace BrewUp.Sales.Facade;

public interface ISalesFacade
{
    Task<string> CreateSalesOrderAsync(CreateSalesOrderJson body, CancellationToken cancellationToken);
    Task<PagedResult<SalesOrderJson>> GetSalesOrdersAsync(int page, int pageSize, CancellationToken cancellationToken);
}